### ISSUE TITLE
How to interpret the results of softmaxloss function?

### ISSUE THREAD

**Issue by noorjugni** [vlfeat/matconvnet#279]
I have a binary classification problem and I put two neurons at the last layer followed by softmaxloss function. But the softmaxloss function returns one value for each of the input examples which doesn't lie between 0-1. So how do we interpret this value (shouldn't this be a probability score)? For example it returns 6.84, 3.73, 2.46 etc for the examples miss-classified. and 2.48, 1.01, 0.002, 0.9 etc for those correctly classified.

**Comment by felixachilles** [comment]
The softmaxLoss function will give you the loss i.e. penalty of the input
samples with regard to their predicted and true label. This number is an
evaluation of the function whose gradient you will backpropagate during
training.

If you want to get the label-probabilites, mostly during testing, exchange
the softmaxLoss layer for a regular softmax layer, both are pre-implemented
in MatConvNet.

2015-10-06 13:27 GMT+02:00 noorjugni <notifications@github.com>:

> I have a binary classification problem and I put two neurons at the last
> layer followed by softmaxloss function. But the softmaxloss function
> returns one value for each of the input examples which doesn't lie between
> 0-1. So how do we interpret this value (shouldn't this be a probability
> score)? For example it returns 6.84, 3.73, 2.46 etc for the examples
> miss-classified. and 2.48, 1.01, 0.002, 0.9 etc for those correctly
> classified.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vlfeat/matconvnet/issues/279>.
>



-- 
-------------------------------------------------------------------
---------------------------------------------------------------

*Technische Universität München**Felix Achilles, M.Sc.** · *Faculty of
Informatics *· *Chair for Computer Aided Medical Procedures
Boltzmannstr. 3 *· *85748 Garching b. München · +49 (0)176 623 18 312
achilles@in.tum.de *· *campar.in.tum.de
------------------------------------------------------
----------------------------------------------------------------------------

**Comment by noorjugni** [comment]
But softmaxloss also has a bug which I think no one checked yet. (Issue 166 and 229)
here is the error I get when I used softmaxloss instead of softmaxlogloss:
Error using  .* 
Matrix dimensions must agree.

Error in vl_nnsoftmax (line 30)
Y = Y .* bsxfun(@minus, dzdY, sum(dzdY .* Y, 3)) ;

Error in vl_simplenn (line 284)
        res(i).dzdx = vl_nnsoftmax(res(i).x, res(i+1).dzdx) ;

Error in cnn_train>process_epoch (line 321)
    res = vl_simplenn(net, im, dzdy, res, ...

Error in cnn_train (line 183)
    [net,stats.train] = process_epoch(netnum, opts, getBatch, epoch, train, learningRate, imdb,
    net) ;
